x-pack/legacy/plugins/uptime/public/components/functional/filter_group/filter_popout.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFieldSearch, EuiFilterSelectItem, EuiPopover, EuiPopoverTitle } from '@elastic/eui';
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { UptimeFilterButton } from './uptime_filter_button';
+import { toggleSelectedItems } from './toggle_selected_item';
+
+export interface FilterPopoverProps {
+  fieldName: string;
+  id: string;
+  isLoading: boolean;
+  items: string[];
+  onFilterFieldChange: (fieldName: string, values: string[]) => void;
+  selectedItems: string[];
+  title: string;
+}
+
+const isItemSelected = (selectedItems: string[], item: string): 'on' | undefined =>
+  selectedItems.find(selected => selected === item) ? 'on' : undefined;
+
+export const FilterPopover = ({
+  fieldName,
+  id,
+  isLoading,
+  items,
+  onFilterFieldChange,
+  selectedItems,
+  title,
+}: FilterPopoverProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [itemsToDisplay, setItemsToDisplay] = useState<string[]>([]);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [tempSelectedItems, setTempSelectedItems] = useState<string[]>(selectedItems);
+
+  useEffect(() => {
+    if (searchQuery !== '') {
+      const toDisplay = items.filter(item => item.indexOf(searchQuery) >= 0);
+      setItemsToDisplay(toDisplay);
+    } else {
+      setItemsToDisplay(items);
+    }
+  }, [searchQuery, items]);
+
+  return (
+    // @ts-ignore zIndex prop is not described in the typing yet
+    <EuiPopover
+      button={
+        <UptimeFilterButton
+          isSelected={tempSelectedItems.length > 0}
+          numFilters={items.length}
+          numActiveFilters={tempSelectedItems.length}
+          onClick={() => setIsOpen(!isOpen)}
+          title={title}
+        />
+      }
+      closePopover={() => {
+        setIsOpen(false);
+        onFilterFieldChange(fieldName, tempSelectedItems);
+      }}
+      id={id}
+      isOpen={isOpen}
+      withTitle
+      zIndex={1000}
```

Comment:
Can we use the $euiZHeader var here?

Reply:
> ID filter: I thought we agreed on dropping the ID filter - is this a misunderstanding?

It was a misunderstanding. Fixed! https://github.com/elastic/kibana/pull/43621/commits/c6453f03bed96a15562b0b9c4fda0cf9ffe53c7f


> If no monitors are found but filters are set - do you think we should update the empty state from No monitors found to something that indicates that filters are set?

Added here. Thoughts on the copy? https://github.com/elastic/kibana/pull/43621/commits/a55b01a416a9d8d3877e1ed21f670c934d224f3f

> Could we use the focus trap for the pop-overs? 

WIP, will update when fixed.